src/games/takesix/game.ts

```diff
@@ -0,0 +1,219 @@
+import { Game, TurnOrder, IGameCtx, INVALID_MOVE, PlayerView } from '@freeboardgame.org/boardgame.io/core';
+import Card from './card';
+import Player from './player';
+
+export interface IG {
+  players: Player[],
+  decks: Card[][],
+  cardOrder: string[],
+  end: boolean,
+}
+
+export function selectCard(G: IG, ctx: IGameCtx, id: number): any {
+  if (id < 0 || id >= G.players[ctx.playerID as any].cards.length) {
+    return INVALID_MOVE;
+  }
+
+  return {
+    ...G,
+    players: Object.values({
+      ...G.players,
+      [ctx.playerID]: {
+        ...G.players[ctx.playerID as any],
+        selectedCard: G.players[ctx.playerID as any].cards.find(
+          (_: Card, index: number) => index === id,
+        ), // Set card as selected
+        cards: G.players[ctx.playerID as any].cards.filter(
+          (_: Card, index: number) => index !== id,
+        ), // Remove card from player's deck
+      },
+    }),
+  };
+}
+
+export function selectDeck(G: IG, ctx: IGameCtx, id: number): any {
+  const lastCards = G.decks
+    .map((deck: Card[]) => deck[deck.length - 1])
+    .sort((a: Card, b: Card) => a.number - b.number);
+  const card = G.players[ctx.playerID as any].selectedCard;
+
+  // Card is lower than every in deck
+  if (card.number < lastCards[0].number) {
+    return {
+      ...G,
+      players: Object.values({
+        ...G.players,
+        [ctx.playerID]: {
+          ...G.players[ctx.playerID as any],
+          penaltyCards: [
+            ...G.players[ctx.playerID as any].penaltyCards,
+            ...G.decks[id], // Move current deck into penalty cards in player's deck
+          ],
+        },
+      }),
+      decks: Object.values({
+        ...G.decks,
+        [id]: [card], // Set selected card as current in selected deck
+      }),
+    };
+  } else {
+    const diffs: number[] = G.decks.map(
+      (deck: Card[]) => card.number - deck[deck.length - 1].number,
+    );
+
+    let min = Number.MAX_SAFE_INTEGER;
+    let minIndex = 0;
+    diffs.forEach((diff, index) => {
+      if (diff > 0 && diff < min) {
+        min = diff;
+        minIndex = index;
+      }
+    });
+
+    if (minIndex !== id) {
+      return INVALID_MOVE;
+    }
+
+    // If card is #6 move all cards from deck to player's hand
+    if (G.decks[id].length === 5) {
+      return {
+        ...G,
+        players: Object.values({
+          ...G.players,
+          [ctx.playerID]: {
+            ...G.players[ctx.playerID as any],
+            penaltyCards: [
+              ...G.players[ctx.playerID as any].penaltyCards,
+              ...G.decks[id],
+            ],
+          },
+        }),
+        decks: Object.values({
+          ...G.decks,
+          [id]: [card],
+        }),
+      };
+    }
```

Comment:
from line 79-96 you reuse this logic both here and from lines 42-58. I think it would be much more readable if you create a separate function "moveToHand" that given G, list of cards and player, returns the new state.